ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PushQueryPublisher.java

```diff
@@ -126,12 +127,14 @@ public synchronized void subscribe(final Flow.Subscriber<Collection<StreamedRow>
               query.getSessionConfig().getConfig(false),
               query.getSessionConfig().getOverrides());
 
+      final ImmutableAnalysis analysis = ksqlEngine
+          .analyzeQueryWithNoOutput(query.getStatement(), query.getStatementText());
```

Comment:
ditto from above, why not pass the analysis in?

Reply:
Oh, so I implemented this advice, but now we actually are computing the analysis twice for _transient_ queries. The reason is that this "push query publisher" has the same constructors whether you're doing a transient or scalable push queries. I tried to make it so we could pass the analysis into the transient query execution, but it turns into a real nightmare without completely disentangling it from persistent queries.

Reply:
Ok, I went ahead and partially walked this back, so for pull queries, we still do it at the top level for pull queries, but for push queries, we do the analysis as soon as we can distinguish between scalable vs. transient push queries.